diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -606,10 +618,11 @@ namespace diff_drive_controller
       }
     }
 
+    // Compute the period estimated from time:
+    const ros::Duration period_estimated = time - time_previous_;
+
     // Compute/Set control period and frequency (desired/expected or real):
-    // @todo also allow to compute the period as time - time_previous_, because
-    // they can be different
-    const double control_period    = control_period_desired_    > 0.0 ? control_period_desired_    : period.toSec();
+    const double control_period    = control_period_desired_    > 0.0 ? control_period_desired_    : (period_from_time_ ? period_estimated.toSec() : period.toSec());
     const double control_frequency = control_frequency_desired_ > 0.0 ? control_frequency_desired_ : 1.0 / control_period;
```

Comment:
Should I check `if (control_period > 0.0)`, and if it's not, then simply skip `return` ?

The `update` method documentation doesn't really says too much about the `period`. It implicitly says it must be monotonic/steady, ie. `period >= 0`. In practice, `period > 0` it's also true. I don't know if there's any situation where `period == 0` is possible. Thoughts?

This is not strictly from this PR, but I think we can discuss about it here.